akd_core/src/types/mod.rs

```diff
@@ -77,40 +113,45 @@ pub enum Direction {
     Left = 0u8,
     /// Right
     Right = 1u8,
-    /// No direction
-    None = u8::MAX,
 }
 
 impl SizeOf for Direction {
     fn size_of(&self) -> usize {
-        // The size of the enum is 24 bytes. The extra 8 bytes are used to store a 64-bit discriminator that is used to identify the variant currently saved in the enum.
+        // The size of the enum is 24 bytes. The extra 8 bytes are used to store a 64-bit
+        // discriminator that is used to identify the variant currently saved in the enum.
         24usize
     }
 }
 
-impl core::convert::TryFrom<u8> for Direction {
-    type Error = String;
-    fn try_from(value: u8) -> Result<Self, Self::Error> {
-        match value {
-            u8::MAX => Ok(Direction::None),
-            0u8 => Ok(Direction::Left),
-            1u8 => Ok(Direction::Right),
-            _ => Err(format!(
-                "Invalid value for direction received: {}. Supported values are 0, 1, {}",
-                value,
-                u8::MAX
-            )),
+impl From<Bit> for Direction {
+    fn from(bit: Bit) -> Self {
+        match bit {
+            Bit::Zero => Self::Left,
+            Bit::One => Self::Right,
         }
     }
 }
 
+impl core::convert::TryFrom<PrefixOrdering> for Direction {
+    type Error = String;
+    fn try_from(prefix_ordering: PrefixOrdering) -> Result<Self, Self::Error> {
+        let direction = match prefix_ordering {
```

Comment:
nit: Generally early-return clauses are helpful in really long fn's but in this case it might be clearer as

```rust
match prefix_ordering {
    PrefixOrdering::WithZero => Ok(Direction::Left),
    PrefixOrdering::WithOne => Ok(Direction::Right),
    _ => Err("Could not convert from PrefixOrdering to Direction".to_string())
}
```